### PR TITLE
upgrade springboot from 2.7.15 to 3.1.3 and java version from 11 to 17

### PR DESCRIPTION
**Pull request by astha-singh-2445**

### CODE REVIEW

File: src/main/java/com/singh/astha/medicinereminder/scheduler/EventReminderCronJob.java
Comment (ShivamNagpal) [inline src/main/java/com/singh/astha/medicinereminder/scheduler/EventReminderCronJob.java:1]: https://github.com/astha-singh-2445/Medicine_Reminder_Spring_App/pull/12/files#diff-f1ec65c9c51f13555c3ce073a4d048acc14c09af0e4eebf9d016e5860184c632R61-R64

- Template Id is hardcoded.
- Why does Title has that values in the map?

Comment (ShivamNagpal) [inline src/main/java/com/singh/astha/medicinereminder/scheduler/EventReminderCronJob.java:1]: https://github.com/astha-singh-2445/Medicine_Reminder_Spring_App/pull/12/files#diff-f1ec65c9c51f13555c3ce073a4d048acc14c09af0e4eebf9d016e5860184c632R74

- Don't call .save() on every iteration.

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -15,6 +15,5 @@ auth.token.public.key=${PUBLIC_KEY}
 server.port=8082
 spring.data.mongodb.uri=${MONGO_URL}
 spring.data.mongodb.auto-index-creation=true
-
 springdoc.api-docs.path=/api-docs
 springdoc.swagger-ui.path=/swagger-ui-custom.html
```
Comment (ShivamNagpal) [inline src/main/resources/application.properties:19]: Add a new line at end of each file

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -15,6 +15,5 @@ auth.token.public.key=${PUBLIC_KEY}
 server.port=8082
```
Comment (ShivamNagpal) [inline src/main/resources/application.properties:15]: Don't hardcode a port in the properties file. You can override this in the Docker while running the application.

File: src/main/java/com/singh/astha/medicinereminder/scheduler/EventReminderCronJob.java
Comment (astha-singh-2445) [reply src/main/java/com/singh/astha/medicinereminder/scheduler/EventReminderCronJob.java:1]: this already cracked as separate task, will raise next pr soon

Comment (astha-singh-2445) [reply src/main/java/com/singh/astha/medicinereminder/scheduler/EventReminderCronJob.java:1]: https://github.com/users/astha-singh-2445/projects/1

Comment (astha-singh-2445) [reply src/main/java/com/singh/astha/medicinereminder/scheduler/EventReminderCronJob.java:1]: https://github.com/users/astha-singh-2445/projects/1

Comment (astha-singh-2445) [reply src/main/java/com/singh/astha/medicinereminder/scheduler/EventReminderCronJob.java:1]: Created an issue for this.
https://github.com/astha-singh-2445/Medicine_Reminder_Spring_App/issues/14

File: src/main/resources/application.properties (hunk 1 above)
Comment (astha-singh-2445) [reply src/main/resources/application.properties]: fixed

**Pull request merged**